src/Models/DataType.php

```diff
@@ -55,6 +55,31 @@ public function deleteRows()
         return $this->rows()->where('delete', 1);
     }
 
+    public function addRelationshipRow($requestData)
```

Comment:
I'm not 100% sure what this is doing (still reading through the code), but this doesn't seem like it belongs in the model.